src/components/modal/v2/parts/Donut.jsx

```diff
@@ -0,0 +1,78 @@
+/** @jsx h */
+/* eslint-disable react/no-unknown-property */
+import { h } from 'preact';
+
+const generateDonutData = (currentNum = 0, numOfPayments) => {
+    return {
+        percentage: (currentNum / numOfPayments) * 100
+    };
+};
+
+const Donut = ({
+    strokeWidth = 3,
+    strokeLinecap = 'round',
+    cx = 21,
+    cy = 21,
+    radius = 15.91549430918954,
+    viewBox = `0 0 ${2 * cx} ${2 * cy}`,
+    style = { fontSize: '0.375rem' },
+    children,
+    segmentStrokeWidth = 4,
+    currentNum = 0,
+    numOfPayments = 4,
+    timeStamp,
+    periodicPayment,
+    qualifying
+}) => {
+    const { percentage } = generateDonutData(currentNum, numOfPayments);
```

Comment:
Consider simplifying this since the function is called once and contains minimal logic which leads to unnecessary jumping around the code while reading through it.
```suggestion
    const percentage = (currentNum / numOfPayments) * 100;
```